### DISCUSSION TITLE
during upgrading from WIX3.14 to HeatWave 4 the launch conditions are not converted and the build fails

### DISCUSSION THREAD

**Discussion by RAppels2022** [wixtoolset/issues#7258] [Questions]
The "old" launch conditions looked like:

    <?xml version="1.0" encoding="utf-8"?>
    <Include>
      
      <?define AdminMessage = "The setup for $(var.ProductName) must be ran under the Administrator account!" ?>

      
      <Property Id="MSIUSEREALADMINDETECTION" Value="0" />
      <Condition Message="$(var.AdminMessage)"><![CDATA[Installed OR Privileged]]></Condition>
    </Include>

for the new Heatwave4 those should look like:

     <Include xmlns="http://wixtoolset.org/schemas/v4/wxs">
      
      <?define AdminMessage = "The setup for $(var.ProductName) must be ran under the Administrator account!" ?>

      
      <Property Id="MSIUSEREALADMINDETECTION" Value="0" />
      <Launch Message="$(var.AdminMessage)" Condition="Installed OR Privileged"/>
    </Include>

but the conversion does not do that. Now I can easily make those adjustments myself (with a lot of help from the online help) but shouldn't the converting process take care of that?

**Comment by RAppels2022** [comment]
I have tried it in VS2019 (latest version 16.11.24 ) and VS2022 (latest version 17.5.1); both have Heatwave version 0.9.5 or in other words 4.0.0-rc3

**Comment by robmen** [comment]
If you run `wix convert` on the original .wxi file, does that get converted correctly?

**Reply by RAppels2022** [reply]
OK, got it done, sorry for the delay but I had to read your wix command-line reference :)
The result is exactly the same

```xml
<Include xmlns="http://wixtoolset.org/schemas/v4/wxs">
  <!-- Check if we are running with elevated rights-->
  <?define AdminMessage = "The setup for $(var.ProductName) must be ran under the Administrator account!" ?>

  <!-- Check that we have elevated permissions if installing -->
  <Property Id="MSIUSEREALADMINDETECTION" Value="0" />
  <Condition Message="$(var.AdminMessage)"><![CDATA[Installed OR Privileged]]></Condition>
</Include>
```

From

```xml
<?xml version="1.0" encoding="utf-8"?>
<Include>
  <!-- Check if we are running with elevated rights-->
  <?define AdminMessage = "The setup for $(var.ProductName) must be ran under the Administrator account!" ?>

  <!-- Check that we have elevated permissions if installing -->
  <Property Id="MSIUSEREALADMINDETECTION" Value="0" />
  <Condition Message="$(var.AdminMessage)"><![CDATA[Installed OR Privileged]]></Condition>
</Include>
```

**Comment by robmen** [comment]
Using the `Include` removed the context for the converter to do its work. In general, I recommend avoiding `<?include?>`.

I'll look and see if `convert` can guess at a solution in this case anyway.

**Comment by RAppels2022** [comment]
I removed the <?include?> directives in the wxi file and the result is unfortunately still the same:
  \
  \<?define AdminMessage = "The setup for $(var.ProductName) must be ran under the Administrator account!" ?>

  \
  \<Property Id="MSIUSEREALADMINDETECTION" Value="0" />
  \<Condition Message="$(var.AdminMessage)"><![CDATA[Installed OR Privileged]]></Condition>

**Reply by robmen** [reply]
What does "I removed the directives in the wxi file" mean? If you removed the code then it (obviously) would not be converted. Did you put it somewhere else? If so, where? Under what element?

**Reply by RAppels2022** [reply]
What I meant was that I removed the starting \<include> and ending \</include> from the wxi file (which did not help either)
```xml
<?xml version="1.0" encoding="utf-8"?>
<Include>
  <!-- Check if we are running with elevated rights-->
  <?define AdminMessage = "The setup for $(var.ProductName) must be ran under the Administrator account!" ?>

  <!-- Check that we have elevated permissions if installing -->
  <Property Id="MSIUSEREALADMINDETECTION" Value="0" />
  <Condition Message="$(var.AdminMessage)"><![CDATA[Installed OR Privileged]]></Condition>
</Include>
```
so the end result is still:
```xml
<Include xmlns="http://wixtoolset.org/schemas/v4/wxs">
  <!-- Check if we are running with elevated rights-->
  <?define AdminMessage = "The setup for $(var.ProductName) must be ran under the Administrator account!" ?>

  <!-- Check that we have elevated permissions if installing -->
  <Property Id="MSIUSEREALADMINDETECTION" Value="0" />
  <Condition Message="$(var.AdminMessage)"><![CDATA[Installed OR Privileged]]></Condition>
</Include>
```

As I said, this is easily fixed manually, your documentation is very precise. Although it would be nice if it is fixed in the end product.
